community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StorePropertyCursor.java

```diff
@@ -53,8 +66,9 @@
     private static final int BITS_BYTE_SIZE = 32;
     private static final int INTERNAL_BYTE_ARRAY_SIZE = 4096;
 
+    private final ByteBuffer pooledBuffer = ByteBuffer.allocate( INTERNAL_BYTE_ARRAY_SIZE );
```

Comment:
I don't like that, I would like to express the fact that it is kept forever, perhaps `cachedBuffer` is better.